lib/solr.ts

```diff
@@ -106,12 +106,14 @@ export class Client {
       ? 'https://'
       : 'http://';
 
-    this.undiciClient = new UndiciClient(
-      `${urlPrefix}${this.options.host}:${this.options.port}`,
-      {
-        connect: this.options.tls,
-      }
-    );
+	  const port = options.noPort ? '' : `:${this.options.port}`;
```

Comment:
why use separate param for that if it could be a simple truthiness check instead?

Reply:
It's just my style of coding. My thought here is that it makes the string template easier to understand. If you prefer, I can change it to this? (Note: we have to account for the colon)...

```
	  this.undiciClient = new UndiciClient(
		`${urlPrefix}${this.options.host}${options.noPort ? '' : `:${this.options.port}`}`,
		{
		  connect: this.options.tls,
		}
	  );
```

Reply:
My point is that `noPort` is completely redundant. You can check for `port` to be set instead.

Reply:
I would prefer to go with your suggestion, but this line will default the port if it isn't provided...
`port: options.port === 0 ? 0 : options.port || '8983'`

If they leave the port off completely in the options, it will be 8983. If they set it to 0, we could do that truthy check to use only the host. I assumed (maybe incorrectly), that the goal of this code was to still use the 0 port like this: 127.0.0.1:0. If you don't intend to use a 0 port, then I agree, we could just leave the :0 off.